From 151b1761de179b7cb25be584fb9e5913dbd2b737 Mon Sep 17 00:00:00 2001
From: Miao Wang <shankerwangmiao@gmail.com>
Date: Sat, 30 Dec 2023 05:15:05 +0800
Subject: [PATCH 1/3] remove c++ dependency

---
 meson.build | 18 +-----------------
 1 file changed, 1 insertion(+), 17 deletions(-)

diff --git a/meson.build b/meson.build
index f7405a2..13feb8d 100644
--- a/meson.build
+++ b/meson.build
@@ -1,4 +1,4 @@
-project('glib', 'c', 'cpp',
+project('glib', 'c',
   version : '2.58.3',
   meson_version : '>= 0.47.0',
   default_options : [
@@ -9,7 +9,6 @@ project('glib', 'c', 'cpp',
 )

 cc = meson.get_compiler('c')
-cxx = meson.get_compiler('cpp')

 cc_can_run = not meson.is_cross_build() or meson.has_exe_wrapper()

@@ -1430,20 +1429,6 @@ if g_have_iso_c_varargs
 #endif''')
 endif

-g_have_iso_cxx_varargs = cxx.compiles('''
-  void some_func (void) {
-    int a(int p1, int p2, int p3);
-    #define call_a(...) a(1,__VA_ARGS__)
-    call_a(2,3);
-  }''', name : 'ISO C99 varargs macros in C++')
-
-if g_have_iso_cxx_varargs
-  glibconfig_conf.set('g_have_iso_cxx_varargs', '''
-#ifdef __cplusplus
-# define G_HAVE_ISO_VARARGS 1
-#endif''')
-endif
-
 g_have_gnuc_varargs = cc.compiles('''
   void some_func (void) {
     int a(int p1, int p2, int p3);
@@ -1974,7 +1959,6 @@ subdir('gio')
 if xgettext.found()
   subdir('po')
 endif
-subdir('tests')

 # Install glib-gettextize executable, if a UNIX-style shell is found
 if have_sh
--
2.39.1
